src/main/java/com/vaadin/recipes/recipe/gridtemplaterendererinteraction/GridTemplateRendererInteraction.java

```diff
@@ -51,6 +60,9 @@ public GridTemplateRendererInteraction() {
                                     "this.getRootNode().getElementById($0).setAttribute('disabled','disabled')",
                                     String.format("button-example-%d", person.getId())
                                 );
+                            disabledMap.put(person.getId(), Boolean.TRUE);
+
+                            UI.getCurrent().access(() -> personGrid.getDataProvider().refreshItem(person));
```

Comment:
Is there a need for `UI.access` here? It is a normal event handler

Reply:
`UI.access` is called in the event handler automatically